### DISCUSSION TITLE
append on useFieldArray

### DISCUSSION THREAD

**Discussion by vnevermore** [react-hook-form/react-hook-form#8485] [Q&A] [answered]
I would like to create an array of inputs with at least 1 input by default. while submitting the form inputs should not be empty.
returned data should be an array with the all the values of inputs

i've created this, and it works great but for some reason, append will not let me add/append new inputs while the previous one is empty.
you can reproduce this issue here ->
https://codesandbox.io/s/restless-moon-fhh27j?file=/src/App.js
it will let you add extra input, only once the previous one is fulfilled with a value , and that's not the behaviour i want.
i would like to be able to add as many as you like in first place.
i can't explain why this is happening

any idea of how to workaround this issue? am i approaching this thing incorrectly? 
thanks in advance to anyone who would spend time on helping me

**Comment by leapful** [comment, marked as answer]
@vnevermore : 

When using "append" function to append new value on field array, there is a "compact" step to filter "undefined" or "null" when merging new value to current field array value and the empty space is also excluded as "compact" is using "Array.filter(Boolean)".

You can update append default value to blank space "  " instead of empty space to make it work as your expectation.

Working demo: https://codesandbox.io/s/romantic-phoebe-4knms2?file=/src/App.js

**Reply by bluebill1049** [reply]
ty @leapful

**Reply by lucasrabiec** [reply]
What about numeric values and providing 0 as a valid value? It will also be filtered by "compact" step.
